File: fibo.erl
-module(fibo).
-export([fibo/1]).

fibo(F) -> fibo(0, 1, F, 0).
fibo(C, L, F, I) when I < F -> out([I, C]), fibo(C + L, C, F, I+1);
fibo(C, _, F, I) when I >= F -> out([I, C]).
out(O) -> io:format("~w: ~w~n", O).

File: fibo.fs

let out i f = printfn "%d: %d" i f
let rec fibr a =
  match a with
  | (c, l, f, i) when i < f -> out i c; fibr(c + l, c, f, i + 1)
  | (c, _, f, i) when i >= f -> out i c
  
let fib f = fibr(0, 1, f, 0)